lib/pacemaker/pcmk_simulate.c

```diff
@@ -389,7 +389,7 @@ pcmk__profile_dir(const char *dir, long long repeat,
         struct stat prop;
         char buffer[FILENAME_MAX];
 
-        out->begin_list(out, NULL, NULL, "Timings");
+        out->begin_list(out, NULL, NULL, "timings");
```

Comment:
`substitutions` is used only if `--xml-substitute` is passed on the command line, which happens by default only for `crm_simulate`, `crmadmin`, `crm_mon`, and `crm_resource`.

That's fine for the `crm_simulate` schema mismatch, since `crm_simulate` is covered... I don't love the idea of outputting the capitalized version for general API output though.

Reply:
Actually the API output and tool XML output do share the same schema (all starting at api-result.rng). I don't remember why `--xml-substitute` is conditional (@clumens?); the output should always conform to the schema.

Reply:
No, I don't remember this anymore and the commit message isn't helpful either.  I'm sure it made sense at the time.  I wonder if I was thinking that all tools should use `--xml-substitute` when converted to using formatted output, and having it conditional was to deal with tools that had been converted to use only the new argument processing but not yet formatted output.  If that's the case, somewhere along the way I/we forgot to add it when we converted tools.

Reply:
We did make it a hidden option, so that it would be considered internal. Let's kill it and always follow the behavior, if it doesn't break existing schemas or regression tests.

Reply:
> all starting at api-result.rng)

Makes sense. I guess I was thinking "nothing specifically ties the `pcmk_profile_dir()` API output to the `crm_simulate*.rng` subset of the `api-result*.rng` schema." Which is still true...

With that being said, to your point, the API output wouldn't validate against the api-result schema in its current state.

> Let's kill it and always follow the behavior, if it doesn't break existing schemas or regression tests.

Sounds good to me.

Reply:
Seems like we don't even need the substitutions array at that point, and instead should replace them all at the source.

Reply:
> If there's other things that need to be validated, it would be helpful to add that somewhere.

I don't quite understand what you mean here. I know that the commands' XML output gets validated against the schema. My points were:
* the C API output isn't tied to a particular sub-schema, only to `api-result*.rng` (but we can't test it directly through a bash script)
* currently it wouldn't even validate against `api-result*.rng` for this particular function

Reply:
What I mean is:  we should have some way of running the C API so that its output can be validated as well.  At the moment, I don't think we're really doing anything like this, unless incidentally one of the API functions is used in a cli tool that's getting run through cts-cli.

Reply:
That sounds reasonable, if perhaps painful and often redundant with the CLI output. Ideally the C API should be tested independently, I agree. It's tested only indirectly via the CLI, and it's not one-to-one.